### PR TITLE
Packaging documentation

### LINKED ISSUE #1237: Document how to add new management packages to the cluster
(This is used to request new product features)

**Describe the feature request**
We need some documentation on how to create a new management package and have it installed/upgraded automatically when a management cluster is created/upgraded.

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [x] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by yharish991**
### What this PR does / why we need it
This PR adds 
1. documentation that provides some guidance on how to add a new package to a package repository and test those generated packages and repo bundles.
2. documentation that provides instructions on how to add a package to a composite package

### Which issue(s) this PR fixes
Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/1237 and https://github.com/vmware-tanzu/tanzu-framework/issues/1885

### Release note
```release-note
Added documentation on how to add a new package to a package repository
Added documentation on how to add a package to a composite package
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: docs/packages/definitions.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Definitions
+
+This document provides definitions of some package terminology used in the Tanzu framework.
+To learn about the Carvel package concepts visit this [page](https://carvel.dev/kapp-controller/docs/latest/packaging)
+
+## Management package
+
+Any package that performs the role of managing and operating a workload load cluster is a Management package.
```
Comment (danniel1205) [inline docs/packages/definitions.md:8]: Question: we might want to remove the extra `load` from `workload load cluster` in above.

Comment (yharish991) [reply docs/packages/definitions.md:8]: thanks for catching that :D

**Comment by jayunit100** [comment]
overall looks like a step forward to me but im not an expert in the packaging docs and what they should be doing

File: docs/packages/add-new-package-to-management-package-repo.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+   ServiceAccount needs to be provided in the PackageInstall CR spec for the management of resources that would be
+   created by it. More info about the kapp-controller security model can be found [here](https://carvel.dev/kapp-controller/docs/latest/security-model/)
+
+## Testing
+
+1. Build the container images for all the packages
+
+   ```shell
+      OCI_REGISTRY=${REGISTRY} make docker all
```
Comment (prkalle) [inline docs/packages/add-new-package-to-management-package-repo.md:105]: should it be `docker-all`

File: docs/packages/dev-workflow.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+   PACKAGE_BUNDLES variable should be comma-separated values and must not contain spaces.
+   Example: PACKAGE_BUNDLES=featuregates,core-management-plugins
+
+6. Push repo bundle
+
+   To push the generated repo bundle to an OCI registry, run:
+
+   ```shell
+      PACKAGE_REPOSITORY=management OCI_REGISTRY=quay.io/hyayiv make push-package-repo-bundle
```
Comment (rajathagasthya) [inline docs/packages/dev-workflow.md:69]: nit: replace `quay.io/hyayiv` with `${REGISTRY}` for consistency.

File: docs/packages/dev-workflow.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+   ```
+
+Follow the below steps to test the artifacts that are generated in previous steps on your local cluster
+
+We can use the `package` plugin to add the repo bundle to the cluster and to install the packages
+
+1. Add package repo bundle to the cluster
```
Comment (rajathagasthya) [inline docs/packages/dev-workflow.md:76]: Does this need a `tanzu login` to the cluster before running `package` commands? If so, a note about that would be helpful.

**Review by rajathagasthya — COMMENTED** [review]
Overall looks good. Just some minor comments.

File: docs/packages/dev-workflow.md (hunk 3 above)
Comment (yharish991) [reply docs/packages/dev-workflow.md:69]: oops, thanks for catching this

File: docs/packages/dev-workflow.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+6. Push repo bundle
+
+   To push the generated repo bundle to an OCI registry, run:
+
+   ```shell
+      PACKAGE_REPOSITORY=management OCI_REGISTRY=quay.io/hyayiv make push-package-repo-bundle
+   ```
+
+Follow the below steps to test the artifacts that are generated in previous steps on your local cluster
```
[the remark this answers is not in the document]
Comment (yharish991) [reply docs/packages/dev-workflow.md:72]: added a notes, let me know what you think.. thanks..

**Review by rajathagasthya — APPROVED** [review]

**Review by codegold79 — APPROVED** [review]
I'll wait for this PR to get merged before I push up a PR for #1782 as my changes will need to be reflected in these docs.

File: docs/packages/README.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Packages
+
+Documentation related to packages and package tooling in Framework repo.
+
+## Table of Contents
+
+* [Definitions](definitions.md)
+* [Add new package to a package repository](add-new-package-to-package-repo.md)
+* [Add PackageInstall to a meta Package](add-packageinstall-to-meta-package.md)
```
Comment (rajathagasthya) [inline docs/packages/README.md:9]: nit: `s/meta/Meta`

File: docs/packages/add-packageinstall-to-meta-package.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+# Add PackageInstall CR to a meta Package
+
+This document provides guidance on how to add a PackageInstall CR to a meta package.
+Check this [doc](./definitions.md#meta-package) to understand some packaging terminology such as
+meta package in the context of this project.
+
+`tkg`, `framework` are some examples of meta package in this repository.
+
+## Steps to add PackageInstall CR to a meta Package
```
Comment (rajathagasthya) [inline docs/packages/add-packageinstall-to-meta-package.md:9]: nit: `s/meta/Meta`

**Review by rajathagasthya — COMMENTED** [review]
Just nits!

File: docs/packages/README.md (hunk 6 above)
Comment (yharish991) [reply docs/packages/README.md]: fixed this

**Review by jmoroski — APPROVED** [review]

**Pull request merged**